### ISSUE TITLE
Does Perception Module Support Volta Architecture (Titan V) ?

### ISSUE THREAD

**Issue by mtyylx** [ApolloAuto/apollo#4233]
We're trying to run apollo perception module with Titan V, but the module will always crash, while the same code running on 1080Ti is fine. 

The error log when running with Titan V is "**check failed error == cudasuccess (8 vs. 0) invalid device function**". This error seem to be related to the customized caffe lib provided by apollo. 

The compilation of apollo is OK. Both the host machine and the docker container are using CUDA 8.0 + 384.111 driver.

Have you guys test apollo on nVIDIA Volta architecture yet?  How can we re-compile the apollo caffe with volta supports?

Log attached:
[perception.in_dev_docker.walle.log.INFO.20180521-194733.3853.log](https://github.com/ApolloAuto/apollo/files/2029330/perception.in_dev_docker.walle.log.INFO.20180521-194733.3853.log)
[perception.in_dev_docker.walle.log.INFO.20180521-200733.4084.log](https://github.com/ApolloAuto/apollo/files/2029331/perception.in_dev_docker.walle.log.INFO.20180521-200733.4084.log)

**Comment by techoe** [comment]
We didn't test Titan V. It may not work on Titan V. Thank you for letting us know. When we have a chance to test the system with Titan V, we will let you know.

Thank you,
Tae Eun

**Comment by natashadsouza** [comment]
@mtyylx have you been successful working with Titan V? If so, it would be great to hear your findings.
Thanks!

**Comment by mtyylx** [comment]
@techoe @natashadsouza Apollo is using its own version of caffe, which is close-sourced. And my guess is that the current caffe.so provided by Apollo is compiled with nvcc flag only up to `sm_61` without a proper PTX version specified. This means the current caffe.so will not support Volta arch, which is `sm_70`.

In Nvidia's Volta Compatibility Guide, it says the GPU in Volta architecture (V100, Titan V, Xavier) will only support CUDA9 + `sm_70` cubin Version or CUDA8 with PTX version specified. 

TL;DR: Apollo can support Volta GPU only if they recompile and release their caffe with PTX version. Otherwise it's impossible.

**Comment by mtyylx** [comment]
> @mtyylx Have you get any response about the cuda9 caffe or have a other solution?

Nop...

**Comment by techoe** [comment]
We do not support Volta architecture yet.
